main.go

```diff
@@ -86,7 +87,15 @@ func main() {
 		}
 	}
 
-	cli, err := NewCli(opts.ProjectId, opts.InstanceId, opts.DatabaseId, opts.Prompt, opts.HistoryFile, cred, os.Stdin, os.Stdout, os.Stderr, opts.Verbose, priority, opts.Role, opts.Endpoint)
+	if opts.DirectedRead != "" {
+		var err error
+		_, err = parseDirectedReadOption(opts.DirectedRead)
+		if err != nil {
+			exitf("Invalid Directed Read option: %v\n", err)
+		}
+	}
```

Comment:
Why not passing parsed result (`*pb.DirectedReadOptions`) into `NewCli`?

For example,
```go
var directedRead *pb.DirectedReadOptions
if opts.DirectedRead != "" {
	var err error
	directedRead, err = parseDirectedReadOption(opts.DirectedRead)
	if err != nil {
		exitf("Invalid Directed Read option: %v\n", err)
	}
}

cli, err := NewCli(....., directedRead)
```

Then we don't need to call `parseDirectedReadOption` again in `NewSession`.

Reply:
It should be! I will fix it.